### DISCUSSION TITLE
Upcoming releases

### DISCUSSION THREAD

**Discussion by legobeat** [depcheck/depcheck#800] [General]
`depcheck` has not seen a release in a while and it would be great to get an idea about how the upgrade path would look like.

First, some observations:

- depcheck is widely used and running in legacy and current systems
- The latest release as of today, [v1.4.3](https://github.com/depcheck/depcheck/releases/tag/1.4.3) was released Jan 2022, over a year ago
- Some well-needed maintenance changes have happened on `master` since and would benefit 1.x users in a drop-in update
  - There are also some dependencies that could be bumped for non-breaking improvements 
  - After a quick look, the only thing that stands out as `semver-major` is addressed in #798 
- depcheck currently indicates Node.js `>=10`. v10 reached EoL 2021-04-30 and has fallen out of use
- If generous compatibility is still desired, it's worth to note that current stable versions of Debian and Ubuntu both have node v12 in distribution repositories

So with that in mind, this would make sense to me:

1. Release `1.5.0` based on current `main` (with any breaking changes addressed/reverted/omitted)
2. Increase `node.engines` to `^12.22.12 || ^14.19 || ^16 || >=18`
3. Release `2.0.0`

Then:
- users on older environments/codebases can still get drop-in replacements addressing `npm audit` issues, security advisories, bug fixes, stability updates and likelier to have less duplicated dependency versions
  - consider also unmaintained libraries currently depdending on some `depcheck@^1`
- `v1` can be fully deprecated and users pointed to `v2`
- when at any point (soon or in the future) the choice is made to make a new breaking `v3` release, `v2` can be kept as a maintenance/support target only receiving security updates and the occasional dependency bump.

**Comment by kevin-donovan-zocdoc** [comment]
I would really like to get some fixes [like this](https://github.com/depcheck/depcheck/pull/724) so a new release would be great. Or at least some timeline on when we can expect the new release

**Comment by rumpl** [comment]
Hey all. 

Indeed depcheck is widely used but depends on me as the only “maintainer”. I haven’t had the time to update the CI away from Travis to gha for releases. Once that is setup releases will come faster. 

Truth is, I have a job and a lot going on in my life right now and all of that takes a lot of time. On top of that I don’t really do any js work these days… depcheck started as a fun hack one night before webpack was a thing, these days there are so many things going on in this place that it’s quite hard to keep up… 

Depcheck had other maintainers in the past but they come and go, I have failed to create a community of people willing to work on it on their free time. 


All this to say, I will try and fix the CI next week and get a release for y’all. I can’t this week, I am on PTO.

**Reply by legobeat** [reply]
@rumpl Thank you for following up and your commitment! I could see if I can propose a PR with the CI switch, if you'd like?

**Reply by rumpl** [reply]
If you have the time and the will that would be awesome

**Reply by a-rothwell** [reply]
Thanks for all you do!
I created a PR to update CI to Github Actions for releases
https://github.com/depcheck/depcheck/pull/803
